app/views/videos/show.html.haml

```diff
@@ -10,5 +10,36 @@
         %p= @video.description
         .actions
           %a.btn.btn-primary Watch Now
-          %a.btn.btn-default + My Queue
+          = link_to "+ My Queue", queue_items_path(video_id: @video.id), method: "POST", class: "btn btn-default"
```

Comment:
I think you should be abler o add video to `+ My Queue` only if it has not been added before. - so maybe add a little `if/unless` condition.